### DISCUSSION TITLE
4.0 Dark Mode

### DISCUSSION THREAD

**Discussion by Stuartemk** [joomla/joomla-cms#32531] [Q&A]
### Steps to reproduce the issue

Is there a way that the Casiopea template integrates by default a dark mode as well as its corresponding switch for users from the frontend?

**Comment by dgrammatiko** [comment]
You will be able to do that once cassiopeia transitions to the new template mode (child templates) with a simple override using https://github.com/vinorodrigues/bootstrap-dark-5 instead of the default bootstrap

**Comment by Stuartemk** [comment]
> You will be able to do that once cassiopeia transitions to the new template mode (child templates) with a simple override using https://github.com/vinorodrigues/bootstrap-dark-5 instead of the default bootstrap

This is excellent news! Only what is not resolved is a togle or switch so that the user can change the template or mode from the frontend. Thank you very much @dgrammatiko

**Comment by dgrammatiko** [comment]
@Stuartemk there are plenty such switchers: https://github.com/dgrammatiko/dark-switch, https://github.com/GoogleChromeLabs/dark-mode-toggle/blob/master/src/dark-mode-toggle.mjs You only have to copy the switch to the child template and edit the HTML to include it, pretty straight forward

**Reply by Stuartemk** [reply]
Most of the administrators of a CMS are not programmers or software developers, that is why we use a CMS and we do not use our own development written from scratch or with a framework such as Laravel or React. Everything is easy for those who already know how to do something, it is very easy to say "You just have to put such code in such a place, download such a library and embed it in such a place, etc." As an administrator and not as a developer or programmer what we can is to activate or deactivate some function through a button, switch or plugin. Thanks for understanding, I really hope that the new version already brings by default the option of a light and a dark theme.

**Reply by himanshu007-creator** [reply]
> Most of the administrators of a CMS are not programmers or software developers, that is why we use a CMS and we do not use our own development written from scratch or with a framework such as Laravel or React. Everything is easy for those who already know how to do something, it is very easy to say "You just have to put such code in such a place, download such a library and embed it in such a place, etc." As an administrator and not as a developer or programmer what we can is to activate or deactivate some function through a button, switch or plugin. Thanks for understanding, I really hope that the new version already brings by default the option of a light and a dark theme.

So true!!. Thats why i think having 2 approaches is good! Either , joomla identifies the system default theme or use manual button to toggle theme. This will increase simplicity.

**Comment by 4digitalworld** [comment]
@Stuartemk I wrote about this earlier - [https://github.com/joomla/joomla-cms/issues/26772](https://github.com/joomla/joomla-cms/issues/26772)

**Comment by himanshu007-creator** [comment]
There can be two things we can try to do:-
1. Make two themes light and dark 
2. User can switch themes by two methods:-
     a) joomla  assigns itself the system default theme set by user
     b) user can change the theme from theme toggler that can be that can be used in the top right corner of admin page.

**Reply by Stuartemk** [reply]
This is how it should and should be.

**Comment by Stuartemk** [comment]
Most of the administrators of a CMS are not programmers or software developers, that is why we use a CMS and we do not use our own development written from scratch or with a framework such as Laravel or React. Everything is easy for those who already know how to do something, it is very easy to say "You just have to put such code in such a place, download such a library and embed it in such a place, etc." As an administrator and not as a developer or programmer what we can is to activate or deactivate some function through a button, switch or plugin. Thanks for understanding, I really hope that the new version already brings by default the option of a light and a dark theme.

**Reply by 4digitalworld** [reply]
5 years have passed and we still don't have a default template with a dark theme.

**Comment by 4digitalworld** [comment]
This is still a very relevant issue, especially since the backend theme has a dark mode, but the frontend theme still doesn't have a dark mode.

**Comment by brettvac** [comment]
For those looking for a theme switcher or dark mode toggle for a front-end template which is light mode only, I created a simple module that you can publish in a visible position that lets you do exactly that. Check it out here: [https://github.com/brettvac/Hoary/](https://github.com/brettvac/Hoary/)
